File: src/dependent/fulfillment/inventory_service.py
class InventoryService:
    def reserve(self, product_id: str, quantity: int) -> str:
        if product_id is None or product_id == "":
            raise ValueError("productId must be provided")
        if quantity <= 0:
            raise ValueError("quantity must be positive")
        return "res-" + str(abs(hash(product_id + str(quantity))))

File: src/dependent/fulfillment/InventoryService.java
package org.example.dependent.fulfillment;

public class InventoryService {
    public String reserve(String productId, int quantity) {
        if (productId == null || productId.isEmpty()) {
            throw new IllegalArgumentException("productId must be provided");
        }
        if (quantity <= 0) {
            throw new IllegalArgumentException("quantity must be positive");
        }
        return "res-" + Math.abs((productId + quantity).hashCode());
    }
}


